flake.nix

```diff
@@ -0,0 +1,82 @@
+{
+  description = "mat-chalmers goes flakes";
+
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  nixConfig.allow-import-from-derivation = true; # cabal2nix uses IFD
+
+  outputs = { self, nixpkgs, flake-utils }:
+    let
+      ghcVer = "ghc945";
+      makeHaskellOverlay = overlay: final: prev: {
+        haskell = prev.haskell // {
+          packages = prev.haskell.packages // {
+            ${ghcVer} = prev.haskell.packages."${ghcVer}".override (oldArgs: {
+              overrides =
+                prev.lib.composeExtensions (oldArgs.overrides or (_: _: { }))
+                  (overlay prev);
+            });
+          };
+        };
+      };
+
+      out = system:
+        let
+          pkgs = import nixpkgs {
+            inherit system;
+            overlays = [ self.overlays.default ];
+            config.allowBroken = true;
+          };
+
+        in
+        {
+          packages = rec {
+            default = mat;
+            mat = pkgs.haskell.packages.${ghcVer}.mat;
+          };
+
+          checks = {
+            inherit (self.packages.${system}) mat;
+          };
+
+          # for debugging
+          # inherit pkgs;
+
+          devShells.default =
+            let haskellPackages = pkgs.haskell.packages.${ghcVer};
+            in
+            haskellPackages.shellFor {
+              packages = p: [ self.packages.${system}.mat ];
+              withHoogle = true;
+              buildInputs =
+                [ pkgs.docker ] ++
+                (with haskellPackages; [
+                  haskell-language-server
```

Comment:
This caused my poor computer to compile hls for eternities.. But I guess we can cachix it